consensus/src/proposal/step.rs

```diff
@@ -63,12 +64,18 @@ impl<T: Operations + 'static, D: Database> ProposalStep<T, D> {
             let failed_certificates =
                 ctx.sv_registry.lock().await.get_failed_certs(iteration);
 
+            let generation_timout = ctx
+                .iter_ctx
+                .get_timeout(node_data::StepName::Proposal)
+                .saturating_sub(Duration::from_secs(1));
```

Comment:
For some corner cases/(localnet experiments) where `generation_timeout` is <= 1 sec, we should have a minimum `generation_timeout` by default.

Reply:
Agree
Related to https://github.com/dusk-network/rusk/pull/1572#discussion_r1532325198 , we can make this value optional on `master`

Reply:
As per @autholykos suggestion, I've changed the timeout to not depend on step_timeout and to be configurable from rusk.toml

Reply:
Could anyone set `generation_timeout` bigger than the minimum Proposal step timeout by mistake? Maybe we can print a warning at least in a such case.